From fc2bc7e230b745dbbd4f4b2cd82e1e5e0c7bf109 Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex.kanavin@gmail.com>
Date: Mon, 11 May 2020 22:19:16 +0000
Subject: [PATCH 2/6] meson.build: correctly set internal vapi dependencies

If they are set as strings, meson will supply the right
arguments to vapigen, but will not set the ninja dependencies
to ensure they get built first, and so races will occur:
https://autobuilder.yoctoproject.org/typhoon/#/builders/64/builds/1881/steps/8/logs/step1b

Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>
(cherry picked from commit 91712a2e131692fa727a0da2868bc23e8df1bf17)
---
 gcr/meson.build | 2 +-
 ui/meson.build  | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/gcr/meson.build b/gcr/meson.build
index 254a933..199452f 100644
--- a/gcr/meson.build
+++ b/gcr/meson.build
@@ -203,7 +203,7 @@ if get_option('introspection')
 
   gcr_vapi = gnome.generate_vapi('gcr-@0@'.format(gcr_major_version),
     sources: gcr_gir[0],
-    packages: [ 'glib-2.0', 'gio-2.0', 'gck-@0@'.format(gck_major_version) ],
+    packages: [ 'glib-2.0', 'gio-2.0', gck_vapi ],
     metadata_dirs: meson.current_source_dir(),
     vapi_dirs: [
       build_root / 'gck',
diff --git a/ui/meson.build b/ui/meson.build
index 5ca3753..477412d 100644
--- a/ui/meson.build
+++ b/ui/meson.build
@@ -174,8 +174,8 @@ if get_option('introspection')
     packages: [
       'glib-2.0',
       'gio-2.0',
-      'gck-@0@'.format(gck_major_version),
-      'gcr-@0@'.format(gcr_major_version),
+      gck_vapi,
+      gcr_vapi,
       'gtk+-3.0'
     ],
     metadata_dirs: meson.current_source_dir(),
-- 
2.20.1

